EFCore.CheckConstraints/Internal/CheckConstraintsConventionSetPlugin.cs

```diff
@@ -1,18 +1,37 @@
+using JetBrains.Annotations;
 using Microsoft.EntityFrameworkCore.Infrastructure;
 using Microsoft.EntityFrameworkCore.Metadata.Conventions;
 using Microsoft.EntityFrameworkCore.Metadata.Conventions.Infrastructure;
-using JetBrains.Annotations;
 using Microsoft.EntityFrameworkCore.Storage;
 
 namespace EFCore.CheckConstraints.Internal
 {
+    /// <summary>
```

Comment:
Please remove these comments - we don't document APIs in Internal namespaces. Same in the other files.

Reply:
Would you like me to remove them all?

I added XML documentation comments, so the user will get IntelliSense when using _CheckConstraints_:

![IntelliSense 1](https://user-images.githubusercontent.com/9283914/110150790-f23a0600-7ddf-11eb-8305-fef898f7c430.png)

![IntelliSense 2](https://user-images.githubusercontent.com/9283914/110150796-f403c980-7ddf-11eb-9ea4-b06fb496ec66.png)
<br/>

In order to avoid numerous warnings, I commented all the code until no more warnings occured.

How do you like me to proceed?

Reply:
Please remove all XML docs from types in Internal namespaces - this code isn't meant to be used by users and is very similar to actual `internal` (or `private`) code.

Reply:
`ValidationCheckConstraintOptionsBuilder` is public, though (see my screenshot above).

Shouldn't `ValidationCheckConstraintOptionsBuilder` then better be moved outside the `internal` folder?

Reply:
You're right - this specific type indeed should be public... Can you please move it out of Internal?

Reply:
But for other cases of types that really should be Internal, we should remove the docs.